docs: add manifest and blog for controlling deployment orders in pipelines (#3527)
aws/copilot-cli: site/content/docs/manifest/pipeline.en.md

The name of an environment to deploy your services to.

<span class="parent-field">stages.</span><a id="stages-approval" href="#stages-approval" class="field">`requires_approval`</a> <span class="type">Boolean</span>  
Optional. Indicates whether to add a manual approval step before the deployment. Defaults to `false`.

<span class="parent-field">stages.</span><a id="stages-deployments" href="#stages-deployments" class="field">`deployments`</a> <span class="type">Map</span>  

> Optional. Control which CloudFormation stacks are going to be deployed and their order.  

Reviewer: ```suggestion
Optional. Control which CloudFormation stacks to deploy and their order.  
```

Author: Done! addressed all the feedback

The `deployments` dependencies are specified in a map of the form:
```yaml
stages:

> - name: test

Reviewer: ```suggestion
  - name: test
```

  deployments:
    <service or job name>:
    <other service or job name>:
      depends_on: [<name>, ...]
```

For example, if your git repository has the following layout:
```
copilot
├── api
│   └── manifest.yml
└── frontend
    └── manifest.yml
```

And you'd like to control the order of your deployments, such that `api` is deployed before `frontend`, then you can configure your stage as follows:
```yaml
stages:

> - name: test

Reviewer: ```suggestion
  - name: test
```

  deployments:
    api:
    frontend:
      depends_on:

>       - api

Reviewer: ```suggestion
        - api
```

Optional. Name of the stack to create or update. Defaults to `<app name>-<stage name>-<deployment name>`.  
For example, if your application is called `demo`, stage name is `test`, and your service name is `frontend` then the stack name will be `demo-test-frontend`.  

<span class="parent-field">stages.deployments.`<name>`.</span><a id="stages-deployments-templatepath" href="#stages-deployments-templatepath" class="field">`template_path`</a> <span class="type">String</span>  
Optional. Path to the CloudFormation template generated during the `build` phase. Defaults to `infrastructure/<deployment name>-<stage name>.yml`.

<span class="parent-field">stages.deployments.`<name>`.</span><a id="stages-deployments-templateconfig" href="#stages-deployments-templatepath" class="field">`template_config`</a> <span class="type">String</span>  
Optional. Path to the CloudFormation template configuration generated during the `build` phase. Defaults to `infrastructure/<deployment name>-<stage name>.params.json`.

<span class="parent-field">stages.</span><a id="stages-test-cmds" href="#stages-test-cmds" class="field">`test_commands`</a> <span class="type">Array of Strings</span>  

> Optional. Commands to run integration or end-to-end tests after deployment. Defaults to no post-deployments validations.

Reviewer: ```suggestion
Optional. Commands to run integration or end-to-end tests after deployment. Defaults to no post-deployment validations.
```
